### ISSUE TITLE
Timelapse Error [Errno 2] 

### ISSUE THREAD

**Issue by anandvip** [mre/timelapse#47]
## System Information

 - Operating system:Mojave
 - Software version:10.14.6

**Comment by ItsThompson** [comment]
Make sure that you have ffmpeg installed (e.g. run brew install ffmpeg)

**Comment by mre** [comment]
Was wondering whether we should bundle `ffmpeg` with the app in order to avoid such scenarios. 🤔

**Comment by ItsThompson** [comment]
Is there anyway you can check if the computer has ffmpeg already installed and then download it?

**Comment by cmangla** [comment]
This user got a notification about ffmpeg only after "stop recording". Maybe we should warn users at startup time, just after the final `print` statement below:

https://github.com/mre/timelapse/blob/a93b77b889e57c3b771edf7b6d135944db9e61fa/timelapse/__main__.py#L146-L156

**Comment by mre** [comment]
oh yeah. That `print(not_found_msg)` should be a `notify`, I guess. Wanna whip up a PR?

**Comment by mre** [comment]
> Is there anyway you can check if the computer has ffmpeg already installed and if there isn't then download it?

Yeah that would be possible. Not sure if it's a bit too invasive, though. Hm...

**Comment by cmangla** [comment]
> oh yeah. That `print(not_found_msg)` should be a `notify`, I guess. Wanna whip up a PR?

I remember some discussion in which we said that `not_found_msg` was too long for a `notify`, am I right?

**Comment by mre** [comment]
Heh, yeah maybe a shorter message would help.

**Comment by dannyalright** [comment]
Hi, new user here.

I'm getting the same error as @anandvip. I did the `brew install ffmpeg` before first-use. I also tried `brew reinstall ffmpeg`. No dice.

Running Catalina 10.15.4 (19E287).

**Comment by mre** [comment]
Welcome @dannyalright,
that's odd. What does `ffmpeg --version` give you on the commandline?

**Comment by cmangla** [comment]
@mre Perhaps we need to add `shell=True` to the line below in `Encoder`:

https://github.com/mre/timelapse/blob/a93b77b889e57c3b771edf7b6d135944db9e61fa/timelapse/encoder.py#L55-L56

Just a guess.

**Comment by dannyalright** [comment]
Thanks, @mre!

`ffmpeg version 4.2.2` is what I get in response to `ffmpeg --version`.

Here's the full response:

```
ffmpeg version 4.2.2 Copyright (c) 2000-2019 the FFmpeg developers
  built with Apple clang version 11.0.3 (clang-1103.0.32.59)
  configuration: --prefix=/usr/local/Cellar/ffmpeg/4.2.2_3 --enable-shared --enable-pthreads --enable-version3 --enable-avresample --cc=clang --host-cflags=-fno-stack-check --host-ldflags= --enable-ffplay --enable-gnutls --enable-gpl --enable-libaom --enable-libbluray --enable-libmp3lame --enable-libopus --enable-librubberband --enable-libsnappy --enable-libsrt --enable-libtesseract --enable-libtheora --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxvid --enable-lzma --enable-libfontconfig --enable-libfreetype --enable-frei0r --enable-libass --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librtmp --enable-libspeex --enable-libsoxr --enable-videotoolbox --disable-libjack --disable-indev=jack
  libavutil      56. 31.100 / 56. 31.100
  libavcodec     58. 54.100 / 58. 54.100
  libavformat    58. 29.100 / 58. 29.100
  libavdevice    58.  8.100 / 58.  8.100
  libavfilter     7. 57.100 /  7. 57.100
  libavresample   4.  0.  0 /  4.  0.  0
  libswscale      5.  5.100 /  5.  5.100
  libswresample   3.  5.100 /  3.  5.100
  libpostproc    55.  5.100 / 55.  5.100
Unrecognized option '-version'.
Error splitting the argument list: Option not found
```

**Comment by anandvip** [comment]
> Make sure that you have ffmpeg installed (e.g. run brew install ffmpeg)

I did, I can see the screen shots in recent files, some odd 1600 screen shots. how ever during installation ffmpeg did instruct to change the path for some of the files, which one? I don't know...

**Comment by anandvip** [comment]
> Was wondering whether we should bundle `ffmpeg` with the app in order to avoid such scenarios. 🤔

Yes, that would be great!

**Comment by digidestination** [comment]
I have the same issue on MacOS Monterey 12.4 any help would be apreciated.

**Comment by mre** [comment]
What does 
```
which ffmpeg
```
say? 

What does this print?

```python
python3 -c 'import shutil; print(shutil.which("ffmpeg"))'
```

**Comment by mre** [comment]
>> Was wondering whether we should bundle ffmpeg with the app in order to avoid such scenarios. 🤔
> Yes, that would be great!

With the new M1 chips we'd end up bundling two versions of ffmpeg I guess: one for Intel Macs and one for arm64. :/ So I don't think we want to bundle ffmpeg after all...

**Comment by digidestination** [comment]
Hello and thank you for your reply, 
Both commands say this: 
/usr/local/bin/ffmpeg
